### PR TITLE
Improve the title test post selection logic

### PR DESCRIPTION
**Pull request by dphiffer**
Fixes title tests (i.e., the problem where non-default test variants wouldn't load).

### CODE REVIEW

File: src/TitleTests.php [hunk 1]
```diff
@@ -294,20 +294,22 @@ function check_variant_url_slugs($request_slug, $query) {
...
 
         // Look up variant URL slugs that match the unknown request slug
-        $post_ids = $wpdb->get_col($wpdb->prepare("
-			SELECT post_id
-			FROM $wpdb->postmeta
-			WHERE meta_key LIKE 'title_variants_%_url_slug'
-			AND meta_value = '%s'
-		", $request_slug), 0);
-		if (empty($post_ids)) {
+        $post_id = $wpdb->get_var($wpdb->prepare("
```
Comment (dphiffer) [inline src/TitleTests.php:297]: I'm changing from an array of IDs (`get_col`) to looking for a single post ID (`get_var`).

File: src/TitleTests.php [hunk 2]
```diff
@@ -294,20 +294,22 @@ function check_variant_url_slugs($request_slug, $query) {
...
+			SELECT pm.post_id
+			FROM $wpdb->postmeta AS pm, $wpdb->posts AS p
+			WHERE pm.meta_key LIKE 'title_variants_%_url_slug'
+			  AND pm.meta_value = '%s'
+              AND pm.post_id = p.ID
+              AND p.post_status = 'publish'
```
Comment (dphiffer) [inline src/TitleTests.php:303]: This join with `wp_posts` is to ensure we don't match against a post revision (which would have status `inherit`).

File: src/TitleTests.php [hunk 3]
```diff
@@ -294,20 +294,22 @@ function check_variant_url_slugs($request_slug, $query) {
...
+		", $request_slug));
+		if (empty($post_id)) {
 			return [];
 		}
 
         // Replace the 'name' query with a list of IDs that have a variant slug
         $query_vars = $query->query_vars;
         unset($query_vars['name']);
-        $query_vars['post__in'] = $post_ids;
+        $query_vars['p'] = $post_id;
```
Comment (dphiffer) [inline src/TitleTests.php:312]: Switch from a search for multiple post IDs to a single ID.

File: src/TitleTests.php [hunk 4]
```diff
@@ -530,8 +532,7 @@ function show_results_summary($post_id) {
      * @return bool
      */
     function is_single() {
-        $is_single = (is_single() && get_post_type() == 'post');
-        return apply_filters('split_tests_is_single', $is_single);
+        return apply_filters('split_tests_is_single', is_single());
```
Comment (dphiffer) [inline src/TitleTests.php:535]: Removing `get_post_type()` from this helper is what fixed the URL slug selection logic for test posts.

File: src/TitleTests.php (hunk 3 above)
Comment (dphiffer) [reply src/TitleTests.php:312]: This was motivated in part by reading the [source code](https://github.com/WordPress/wordpress-develop/blob/6.7/src/wp-includes/query.php#L729-L738) for `is_single()`, which calls `$wp_query->is_single()`, which checks for a few query params including `p`: https://github.com/WordPress/wordpress-develop/blob/6.7/src/wp-includes/class-wp-query.php#L874

**Review by BatMiles — APPROVED** [review]
🚀

**Pull request merged**